src/misc/U2fClient.js

```diff
@@ -79,7 +79,7 @@ export class U2fClient {
 			], [], resolve, TIMEOUT)
 		})
 		const response = this._handleRawResponse(rawResponse)
```

Comment:
This seems wrong.
`response` is not a Promise, (check `_handleRawResponse` and maybe improve type of it if you have time?)
`rawResponse` *is* a promise because we create it on the previous line.

Reply:
rawResponse seems to be of type


```js
u2f.Error|u2f.RegisterResponse
```

with

```js
/**
 * Data object for a registration response.
 * @typedef {{
 *   version: string,
 *   keyHandle: string,
 *   transports: Transports,
 *   appId: string
 * }}
 */
u2f.RegisterResponse;
```

```js
/**
 * An error object for responses
 * @typedef {{
 *   errorCode: u2f.ErrorCodes,
 *   errorMessage: ?string
 * }}
 */
u2f.Error;
```

with

```js
/**
 * Response status codes
 * @const
 * @enum {number}
 */
u2f.ErrorCodes = {
	'OK': 0,
	'OTHER_ERROR': 1,
	'BAD_REQUEST': 2,
	'CONFIGURATION_UNSUPPORTED': 3,
	'DEVICE_INELIGIBLE': 4,
	'TIMEOUT': 5
};
```

Reply:
Yes, you are right, it seems wrong. I am a bit puzzled that it did work though... I'm on it!

Reply:
I guess we could also start typing the `u2f-api.js` wrapper, but i am not sure that is helpful at all..